### PR TITLE
Misc Improvements to Products & Page Attachments appearance

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

**Style Panel**

* Hides Link & Animation tabs for products — they do not support them
* Updates selection tab icon for product — uses existing shopping bag icon
* Displays the product dropdown (or a disabled input field for mixed selection) in the sidebar for product selection — instead of a TODO
    * **Note:** While this is somewhat redundant with the dropdown in the floating menu, it's better than having nothing here. In the future we will revisit this.

**Shopping / Page Attachment**

* Adds a scrim to the shopping attachment when using the light theme
* Improves appearance when canvas is smaller — previously they used hardcoded px values

## User-facing changes

**Products Style Sidebar**

Before:
Screenshot 2022-05-24 at 15 34 55

After:

after

Screenshot 2022-05-24 at 12 46 50

**Shopping Attachment**

New scrim for light theme:

scrim

**Shopping and Page Attachments**

Improved appearance when canvas is smaller:

Screenshot 2022-05-24 at 15 15 28
Screenshot 2022-05-24 at 15 08 04

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

**Shopping**

1. Verify that the Link & Animation panels are hidden when a product is selected on the canvas
2. In the Style tab, there should be a product dropdown when a product is selected on the canvas — or a disabled input field saying "Mixed" when multiple different products are selected
3. The "Shop Now" link at the bottom should now more closely match the appearance when previewing on the frontend
4. The appearance of the "Shop Now" link at the bottom should be improved when the canvas is smaller (i.e. when resizing the window)

**Page Attachment**

1. The appearance of the  "Learn More" link at the bottom should be improved when the canvas is smaller (i.e. when resizing the window)

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11597

### CODE REVIEW

File: packages/design-system/src/components/datalist/datalist.js [hunk 1]
```diff
@@ -198,7 +199,7 @@ const Datalist = forwardRef(function Datalist(
   const selectedOption = primaryOptions.find(({ id }) => id === selectedId);
   // In case of isInline, the list is displayed with 'absolute' positioning instead of using a separate popup.
   return (
-    <Container>
+    <Container className={className}>
```
Comment (swissspidy) [inline packages/design-system/src/components/datalist/datalist.js:204]: This is for making styled-components work, as it requires the `className` prop to be passed to the component when doing `styled(DataList.DropDown)`

File: packages/story-editor/src/components/canvas/frameElement.js [hunk 2]
```diff
@@ -105,13 +105,12 @@ const EmptyFrame = styled.div`
...
+  /* translators: %s: Ctrl+Alt+P keyboard shortcut. */
   __(
-    'To exit the canvas area, press Escape. Press Tab to move to the next group or element. To enter floating menu, press %1$s %2$s p.',
+    'To exit the canvas area, press Escape. Press Tab to move to the next group or element. To enter floating menu, press %s.',
     'web-stories'
   ),
-  prettifyShortcut('ctrl'),
-  prettifyShortcut('alt')
+  prettifyShortcut('ctrl+alt+p')
 );
```
Comment (swissspidy) [inline packages/story-editor/src/components/canvas/frameElement.js:114]: Unrelated change but something that I just spotted along the way.

**Review by spacedmonkey — DISMISSED** [review]

**Review by merapi — APPROVED** [review]
Not a big deal, but something is off with the Style dropdown?

https://user-images.githubusercontent.com/21036224/171200816-a0a364b1-fea3-40b2-b362-e296b70a713a.mov

**Comment by swissspidy** [comment]
@merapi Probably because the Style panel uses `pushUpdate` and changes are only actually applied when leaving the panel? Is there a way to work around this?

**Review by timarney — APPROVED** [review]

**Pull request merged**